tests/integration/clientv3/util.go

```diff
@@ -119,3 +125,62 @@ func populateDataIntoCluster(t *testing.T, cluster *integration2.Cluster, numKey
 		}
 	}
 }
+
+// guaranteeRaftAppliedIndexAdvance ensures raftLog.applied is the same as appliedIndex of etcdserver.EtcdServer
+// it should only be used between each raft proposal.
+func guaranteeRaftAppliedIndexAdvance(t *testing.T, gRPCURL string, clientURL url.URL) {
+	t.Logf("ensuring applied index advance, (gRPCURL %s), (clientURL %v)", gRPCURL, clientURL)
+	cc, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{gRPCURL},
+		DialTimeout: 5 * time.Second,
+	})
+	require.NoError(t, err)
+	defer cc.Close()
+	cresp, err := cc.Status(context.TODO(), gRPCURL)
+	require.NoError(t, err)
+	appliedIndex := cresp.RaftAppliedIndex
+
+	require.Eventually(t, func() bool {
+		tr, err := transport.NewTransport(transport.TLSInfo{}, 5*time.Second)
+		require.NoError(t, err)
+		clientURL.Path = "/debug/vars"
+		response, err := tr.RoundTrip(&http.Request{
+			Header: make(http.Header),
+			Method: http.MethodGet,
+			URL:    &clientURL,
+		})
+
+		body, err := io.ReadAll(response.Body)
+		require.NoError(t, err)
+		var resp map[string]interface{}
+		require.NoError(t, json.Unmarshal(body, &resp))
+		var raftStatus raftStatus
+
+		for key, value := range resp {
+			if key == "raft.status" {
+				rs, err := json.Marshal(value)
+				require.NoError(t, err)
+				t.Logf("marshalled raft status is %s", string(rs))
+				require.NoError(t, json.Unmarshal(rs, &raftStatus))
+				break
+			}
+		}
+		return raftStatus.Applied == appliedIndex
+	}, 5*time.Second, 20*time.Millisecond)
+}
+
+// raftStatus should be in sync with raft.Status
```

Comment:
is it possible to reuse raft.Status without defining raftStatus?

Reply:
Unfortunately not. The raft.Status does not provide an Unmarshal method. I think it can be raised as a follow up PR in etcd-it/raft.

Reply:
I do not get this, why can't reuse `raft.Status`? Your `raftStatus` also doesn't have an `Unmarshal` method.